client/clients/router/client.go

```diff
@@ -367,22 +397,53 @@ func (c *Cli) updateConnection(ctx context.Context) {
 	cc, url := c.getLeaderClientConn()
 	if cc == nil || len(url) == 0 {
 		log.Warn("[router] got an invalid leader client connection", zap.String("url", url))
-		return
 	}
 	if c.conCtxMgr.Exist(url) {
```

Comment:
If the url is invalid, do we still need to check Exist?

Reply:
Refined this check.